lib/mockito.dart

```diff
@@ -48,6 +53,119 @@ class Mock {
   String toString() => _givenName != null ? _givenName : runtimeType.toString();
 }
 
+// Return a new [Invocation], reconstituted from [invocation], [_typedArgs],
+// and [_typedNamedArgs].
+Invocation _reconstituteInvocation(Invocation invocation) {
+  var newInvocation = new FakeInvocation(invocation);
+  _typedArgs.clear();
+  _typedNamedArgs.clear();
+  return newInvocation;
+}
+
+/// An Invocation class with a functional constructor.
+class FakeInvocation extends Invocation {
+  final Symbol memberName;
+  final Map<Symbol, dynamic> namedArguments;
+  final List<dynamic> positionalArguments;
+  final bool isGetter;
+  final bool isMethod;
+  final bool isSetter;
+
+  factory FakeInvocation(Invocation invocation) {
+    if (_typedArgs.isEmpty && _typedNamedArgs.isEmpty) {
+      return null;
```

Comment:
Since this is only called when _typedArgs. or _typedNamedArgs is not empty, why not throw in this case?